src/wallet/spend.cpp

```diff
@@ -692,7 +708,24 @@ util::Result<SelectionResult> AutomaticCoinSelection(const CWallet& wallet, Coin
         }
 
         // Group outputs and map them by coin eligibility filter
-        FilteredOutputGroups filtered_groups = GroupOutputs(wallet, available_coins, coin_selection_params, ordered_filters);
+        std::vector<OutputGroup> discarded_groups;
+        FilteredOutputGroups filtered_groups = GroupOutputs(wallet, available_coins, coin_selection_params, ordered_filters, discarded_groups);
+
+        // Check if we still have enough balance after applying filters (some coins might be discarded)
+        CAmount total_discarded = 0;
+        CAmount total_unconf_long_chain = 0;
```

Comment:
q: do I understand correctly with current filters `total_discarded` is always equal to `total_unconf_long_chain`?

Reply:
Nop. Could also have zero conf change (or send-to-self) if the wallet's `m_spend_zero_conf_change` flag is disabled. In other words, unconfirmed coins accepted by the mempool limits (ancestors/descendants < 25).